goembed.go

```diff
@@ -21,41 +22,85 @@ var (
 func main() {
 	flag.Parse()
 
-	raw, err := ioutil.ReadAll(os.Stdin)
+	fmt.Printf("package %s\n\n", *packageFlag)
+
+	if *gzipFlag {
+		if err := gzipPrologue.Execute(os.Stdout, struct {
```

Comment:
the args are too big here. Pull the statement out of the `if`, and just use a map; the trivial loss of type safety doesn't matter here.

```go
err := gzipPrologue.Execute(os.Stdout, map[string]interface{}{
  "Args": ...,
  ...
})
if err != nil { log.Fatal(err) }
```